photon-api/src/main/scala/com/linkedin/photon/ml/model/FixedEffectModel.scala

```diff
@@ -115,22 +105,21 @@ object FixedEffectModel {
   /**
    * Compute the scores for the dataset.
    *
-   * @param dataPoints The dataset to score
+   * @param dataset The dataset to score
    * @param modelBroadcast The model to use for scoring
    * @param featureShardId The feature shard id
    * @return The scores
    */
-  private def score[T, V](
-      dataPoints: RDD[(UniqueSampleId, GameDatum)],
+  private def score(
+      dataset: DataFrame,
       modelBroadcast: Broadcast[GeneralizedLinearModel],
-      featureShardId: FeatureShardId,
-      toScore: (GameDatum, Double) => T,
-      toResult: RDD[(UniqueSampleId, T)] => V): V = {
+      featureShardId: FeatureShardId): DataFrame = {
 
-    val scores = dataPoints.mapValues { gameDatum =>
-      toScore(gameDatum, modelBroadcast.value.computeScore(gameDatum.featureShardContainer(featureShardId)))
-    }
+    val cofs = VectorUtils.breezeToMl(modelBroadcast.value.coefficients.means)
+    val scores = dataset
+      .withColumn(DataConst.SCORE, GeneralizedLinearModel.scoreUdf(lit(cofs), col(featureShardId)))
+      .select(Constants.UNIQUE_SAMPLE_ID, DataConst.SCORE)
```

Comment:
See comments for the `RandomEffectModel` - don't drop scores. Not certain yet, but we may need the score function to become a `GameModel` exclusive function.